services/src/main/java/org/keycloak/broker/saml/SAMLIdentityProviderFactory.java

```diff
@@ -99,6 +102,10 @@ public SAMLIdentityProviderConfig createConfig() {
 
                 if (idpDescriptor != null) {
                     SAMLIdentityProviderConfig samlIdentityProviderConfig = new SAMLIdentityProviderConfig();
+                     
+                    // find default locale
+                    String lang = (String) session.getAttribute("locale");
```

Comment:
This is not the way to obtain the default locale. Session is not intended to store variables like this.